### PR TITLE
Adding examples for Encoding Module 

### PR DESCRIPTION
**Pull request by datacubeR**
Hi @solegalli 

I just added examples for the encoding module. 
* I would like your feedback on this especially because I'm not sure how to test this examples are ok. Is there any way to render a dev documentation to see it?
* Second, scikit-learn shows some kind of glimpse of the answer, but they use `np.arrays` so it's a bit more concise than showing a glimpse of a dataframe, what do you think about that?

Please let me know if we are OK with this before start working on other modules. 

Finally, do you think it could be a good a idea to have some notebooks for every module with this examples?

Thanks, 

Alfonso

### CODE REVIEW

File: feature_engine/encoding/count_frequency.py [hunk 1]
```diff
@@ -119,6 +119,16 @@ class CountFrequencyEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
...
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.encoding import CountFrequencyEncoder
+    >>> X = pd.DataFrame(dict(a = [1,2,3,4], b = ["c", "a", "b", "c"]))
+    >>> cf = CountFrequencyEncoder(encoding_method='count')
+    >>> cf.fit(X)
+    >>> cf.transform(X)
```
Comment (solegalli) [inline feature_engine/encoding/count_frequency.py:131]: good be great to show the resulting dataframe.

File: feature_engine/encoding/decision_tree.py [hunk 2]
```diff
@@ -156,6 +156,17 @@ class DecisionTreeEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
...
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.encoding import DecisionTreeEncoder
+    >>> X = pd.DataFrame(dict(a = [1,2,3,4,5], b = ["b", "b", "b", "a", "a"]))
+    >>> y = pd.Series([2.2,4, 1.5, 3.2, 1.1])
+    >>> ohe = DecisionTreeEncoder(cv=2)
+    >>> ohe.fit(X, y)
+    >>> ohe.transform(X)
```
Comment (solegalli) [inline feature_engine/encoding/decision_tree.py:169]: good be great to show the resulting df, and also maybe not call the encoder ohe?

File: feature_engine/encoding/mean_encoding.py [hunk 3]
```diff
@@ -146,6 +146,17 @@ class MeanEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
     .. [1] Micci-Barreca D. "A Preprocessing Scheme for High-Cardinality Categorical
        Attributes in Classification and Prediction Problems". ACM SIGKDD Explorations
        Newsletter, 2001. https://dl.acm.org/citation.cfm?id=507538
+
+        Examples
```
Comment (solegalli) [inline feature_engine/encoding/mean_encoding.py:150]: indentation

**Review by solegalli — COMMENTED** [review]
Hi @datacubeR 

great job, thank you!

[Here are guidelines](https://feature-engine.readthedocs.io/en/latest/contribute/contribute_docs.html#build-the-documentation) to build the docs. After these 2 commands, you should see a folder called "build" and in there, you can double click on the file called "index" to render the docs in a browser.

I think, the only thing I would add is a display of the transformed dataframe? I think that may help users to visually understand what the output of the encoding is meant to be.

We do have notebooks with examples in this repo: https://github.com/feature-engine/feature-engine-examples

And we link to them from the user guide. Not sure if they need updating after the various releases of Feature-engine.

Also, let's make one PR per module (this was probably your idea already?). It helps with the review and also to link with other PRs that may modify some of the same files.

Thanks a lot for your contribution!

**Comment by datacubeR** [comment]
@solegalli 
I rendered a local vervion of the docs to check how it is being rendered. I get this:

image
So it can be seen that `>>>` doesn't work like in Scikit-Learn ([see this](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html)) allowing to hide the prompts. 

I found a sphinx extension that allows something like this: https://github.com/jurasofish/sphinx-toggleprompt , but I'm not proficient enough in sphinx to apply a change like this. And I thought this is a major change. What do you think, is it worthy to apply something like this or should I get rid of `>>>`?

**Comment by solegalli** [comment]
Hi @datacubeR 

Yes, I am aware that you need to add some functionality to be able to remove the >>>.

Could you create an issue with the link?

I'd say, leave the >>> in, and we'll take care of adding the functionality to remove them, in a different PR.

Thank you!

**Comment by solegalli** [comment]
Or actually, when you are done with the code for this PR, I can try adding the functionality, before merging, because it seems fairly straightforward.

Let me know when you finish here, and I can give it a go.

**Comment by datacubeR** [comment]
@solegalli this one is ready for review. If OK, I can move on with other modules (in different PRs).

**Comment by solegalli** [comment]
Hi @datacubeR 

Yes, this is good to merge.

I made a PR to your repo: https://github.com/datacubeR/feature_engine/pull/1 to add the button to remove the `>>>`

If you merge that PR, then I go ahead and merge here.

I look forward to your examples for the other modules :)

Thanks a lot!

**Comment by datacubeR** [comment]
Hi @solegalli quick question, to check the resulting copy button. My local branch is now 1 commit behind because of this PR merge. What is the proper way  to update local branch with this updated one? Is it just a rebase? how should that be applied?

**Comment by solegalli** [comment]
> 

You mean to sync your feature branch with the PR I made to your repo?

I think, after you merge the PR, if you do `git pull origin feature-branch-name` should work. But you won't need that anymore, because I am merging as we speak :)

Thanks a lot for the contribution @datacubeR !

**Pull request merged**